pkg/drivers/generic/generic.go

```diff
@@ -442,10 +442,19 @@ func (d *Generic) Insert(ctx context.Context, key string, create, delete bool, c
 	for i := uint(0); i < 20; i++ {
 		row := d.queryRow(ctx, d.InsertSQL, key, cVal, dVal, createRevision, previousRevision, ttl, value, prevValue)
 		err = row.Scan(&id)
+
 		if err != nil && d.InsertRetry != nil && d.InsertRetry(err) {
+			logrus.Warnf("retriable insert error for key %v: %v", key, err)
+			metrics.InsertErrorsTotal.WithLabelValues(key, "true").Inc()
```

Comment:
I would recommend against including the key in this metric. Log it if you want, but including it in metrics can lead to excessively large metric cardinality. Ref: https://prometheus.io/docs/practices/naming/#labels

> Do not use labels to store dimensions with high cardinality (many different label values), such as user IDs, email addresses, or other unbounded sets of values.